besu/src/main/java/org/hyperledger/besu/services/BlockchainServiceImpl.java

```diff
@@ -135,6 +142,17 @@ public Optional<Hash> getFinalizedBlock() {
     return blockchain.getFinalized();
   }
 
+  @Override
+  public void setFinalizedBlock(final Hash blockHash) {
+    if (genesisConfigOptionsSupplier != null) {
```

Comment:
Just checking the genesis isn't going to work for chains that transition to POS.

You could use the protocolSchedule to check the chain is POS for given blockHash by doing something like
```protocolSchedule.getByBlockHeader(blockchain.getBlockHeader(blockHash).get()).isPoS()```

Reply:
done.